code/modules/chemistry/Chemistry-Tools.dm

```diff
@@ -817,11 +825,85 @@ proc/ui_describe_reagents(atom/A)
 				src.add_reagents_to_containers(reagent, divided_amount - remaining_container_space, sdata, temp_new, donotreact, donotupdate)  //...then run the whole proc again with the remaining reagent, evenly distributing to remaining containers
 			else
 				container.reagents.add_reagent(reagent, divided_amount, sdata, temp_new, donotreact, donotupdate)
-
 	disposing()
 		src.remove_container()
 		. = ..()
 
+	fractional
+		var/container_count = 4
+		var/container_order[4]
+		desc = "A set of glass tubes, conveniently capable of splitting the outputs of more advanced reactions. Can be hooked up to many types of containers."
+		icon_state = "condenser_fractional"
+		update_icon()
+			src.UpdateOverlays(null, "fluid_image")
+			if (reagents.total_volume)
+				var/fluid_state = round(clamp((src.reagents.total_volume / src.reagents.maximum_volume * 5 + 1), 1, 5))
+				if (!src.fluid_image)
+					src.fluid_image = image(src.icon, "fluid-condenser_fractional[fluid_state]", -1)
+				else
+					src.fluid_image.icon_state = "fluid-condenser_fractional[fluid_state]"
+				var/datum/color/average = reagents.get_average_color()
+				src.fluid_image.color = average.to_rgba()
+				src.UpdateOverlays(src.fluid_image, "fluid_image")
```

Comment:
this looks like it's a duplicate of the parent proc with different `icon_state`s? If so it should just use a var instead